### PR TITLE
Don't HTML escape source code blocks in Markdown

### LINKED ISSUE #2252: Code blocks in Markdown output don't need HTML escapes
Dartdoc generates fenced blocks for code, which should be rendered as-is. Using HTML escapes within these fenced blocks results in code that can be diffcult to read.

Screenshot from 2020-07-07 10-12-13

### PR DESCRIPTION
**Pull request by jdkoren**
Adds a renderer in between to control whether or not we apply HTML
escapes to source code blocks.

Fixes #2252

### CODE REVIEW

File: lib/src/render/source_code_renderer.dart [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
```
Comment (srawlins) [inline lib/src/render/source_code_renderer.dart:1]: This should be 2020.

File: lib/src/render/source_code_renderer.dart [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:convert';
+
+abstract class SourceCodeRenderer {
```
Comment (srawlins) [inline lib/src/render/source_code_renderer.dart:7]: I think the implementations or the abstract class should have a doc comment.

File: lib/src/render/source_code_renderer.dart (hunk 1 above)
Comment (jdkoren) [reply lib/src/render/source_code_renderer.dart]: Fixed

File: lib/src/render/source_code_renderer.dart (hunk 2 above)
Comment (jdkoren) [reply lib/src/render/source_code_renderer.dart:8]: Thanks, added some in next commit

File: lib/src/render/source_code_renderer.dart [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  String renderSourceCode(String source) => source;
+}
+
+/// [SourceCodeRenderer] that escapes characters for HTML.
+class SourceCodeRendererHtml extends SourceCodeRenderer {
+  @override
+  String renderSourceCode(String source) {
+    return (const HtmlEscape()).convert(source);
+  }
+}
```
Comment (devoncarew) [inline lib/src/render/source_code_renderer.dart:23]: As a drive by comment, I'd `dartfmt` this file.

Comment (jdkoren) [reply lib/src/render/source_code_renderer.dart]: Indeed, there were a few files flagged by dartfmt.

**Pull request merged**